src/main.rs

```diff
@@ -247,15 +247,14 @@ fn main() -> Result<(), Box<dyn Error>> {
 
     let exts = &cli.extensions;
 
-    let mut extension = ExtensionOptionsBuilder::default();
-    extension
+    let extension = ExtensionOptions::builder()
         .strikethrough(exts.contains(&Extension::Strikethrough) || cli.gfm)
         .tagfilter(exts.contains(&Extension::Tagfilter) || cli.gfm)
         .table(exts.contains(&Extension::Table) || cli.gfm)
         .autolink(exts.contains(&Extension::Autolink) || cli.gfm)
         .tasklist(exts.contains(&Extension::Tasklist) || cli.gfm)
         .superscript(exts.contains(&Extension::Superscript))
-        .header_ids(cli.header_ids)
+        .maybe_header_ids(cli.header_ids)
```

Comment:
For optional fields (or fields with `#[builder(default)]`) `bon` generates two setters:
- `<field>(value: T)`
- `maybe_<field>(value: Option<T>)`

This is desingned this way so that [changing a required field to optional is compatible](https://elastio.github.io/bon/guide/compatibility#making-a-required-member-optional), and it also simplifies the syntax for optional members by removing the need to wrap the value with `Some()` using the `<field>` setter